crypto/context.c

```diff
@@ -177,6 +178,9 @@ static int context_init(OSSL_LIB_CTX *ctx)
     ctx->self_test_cb = ossl_self_test_set_callback_new(ctx);
     if (ctx->self_test_cb == NULL)
         goto err;
+    ctx->indicator_cb = ossl_indicator_set_callback_new(ctx);
```

Comment:
I dont think it makes much difference get_indicator_callback() would then just not do. ossl_lib_ctx_get_data().
Every other field in libctx works this way, so I was trying to keep it the same.

Reply:
The callback cannot live inside a lib ctx, it has be global and singular.

The FIPS provider's lib ctx structure is different to the core's.  Hence, the FIPS provider cannot access anything from a core lib ctx, it can only pass the pointer back to the core.

The next point would be about adding a callback to the core that gets the appropriate data from the core's lib ctx.  This would be possible but it would be quite difficult.  The FIPS provider has it's own internal lib ctx which it uses to fetch algorithms from itself.  I.e. a KDF that needs a digest will fetch it from the FIPS provider's internal lib ctx and not from the caller's core lib ctx.  This cascades when, e.g., a KDF uses a MAC which in turn uses a cipher.